app_dart/lib/src/request_handling/authentication.dart

```diff
@@ -135,68 +139,99 @@ class AuthenticationProvider {
     } else if (isCron) {
       // Authenticate cron requests that are not agents.
       return AuthenticatedContext._(clientContext: clientContext);
-    } else if (idToken != null) {
-      // Authenticate as a signed-in Google account via OAuth id token.
-      final HttpClient client = _httpClientProvider();
-      try {
-        final HttpClientRequest verifyTokenRequest = await client.getUrl(Uri.https(
-          'oauth2.googleapis.com',
-          '/tokeninfo',
-          <String, String>{
-            'id_token': idToken,
-          },
-        ));
-        final HttpClientResponse verifyTokenResponse = await verifyTokenRequest.close();
-
-        if (verifyTokenResponse.statusCode != HttpStatus.ok) {
-          final String body = await utf8.decodeStream(verifyTokenResponse);
-          log.warning('Token verification failed: ${verifyTokenResponse.statusCode}; $body');
-          throw const Unauthenticated('Invalid ID token');
-        }
-
-        final String tokenJson = await utf8.decodeStream(verifyTokenResponse);
-        TokenInfo token;
+    } else if (idTokenFromCookie != null || idTokenFromHeader != null) {
+      /// There are two possible sources for an id token:
+      ///
+      /// 1. Angular Dart app sends it as a Cookie
+      /// 2. Flutter app sends it as an HTTP header
+      ///
+      /// The Flutter application sends both since the client cookies are forwarded
+      /// in the request. One of these may be unauthenticated, so we need to try
+      /// authenticating both.
```

Comment:
The criterion is unclear, is it "the user is authenticated as long as one id token is valid"?

Reply:
How does "As long as one of these two id tokens are authenticated, the request is authenticated." sound?